Android/quickstart-kotlin/app/src/main/java/com/example/immersivereadersdk/MainActivity.kt

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.example.immersivereadersdk
+
+import android.app.Activity
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.webkit.CookieManager
+import android.webkit.WebView
+import android.widget.Button
+import android.webkit.WebViewClient
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.google.gson.*
+import io.github.cdimascio.dotenv.dotenv
+import java.io.IOException
+import java.io.*
+import java.net.HttpURLConnection
+import java.net.HttpURLConnection.HTTP_OK
+import java.net.URL
+import kotlinx.coroutines.*
+import org.json.JSONObject
+import java.util.*
+import kotlin.collections.HashMap
+import kotlin.reflect.KFunction0
+
+// This sample app uses the Dotenv is a module that loads environment variables from a .env file to better manage secrets.
+// https://github.com/cdimascio/java-dotenv
+// Be sure to add a "env" file to the /assets folder
+// instead of '.env', use 'env'
+
+class MainActivity : AppCompatActivity() {
+
+    private val dotEnv = dotenv {
+        directory = "/assets"
+        filename = "env"
+        ignoreIfMalformed = true
+        ignoreIfMissing = true
+    }
+
+    private lateinit var contextualWebView: WebView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        this.supportActionBar!!.hide()
+        setContentView(R.layout.activity_main)
+        val immersiveReaderButton = findViewById<Button>(R.id.LaunchImmersiveReaderButton)
+        immersiveReaderButton.setOnClickListener { GlobalScope.launch { handleLoadImmersiveReaderWebView() } }
+    }
+
+    private suspend fun handleLoadImmersiveReaderWebView() {
+        val exampleActivity = this
+        val subdomain = dotEnv["SUBDOMAIN"]
+        val irTitle = findViewById<TextView>(R.id.Title)
+        val irText1 = findViewById<TextView>(R.id.Content1)
+        val irText2 = findViewById<TextView>(R.id.Content2)
+        val appUtilities = AppUtilities(exampleActivity)
+
+        // The content of the request that's shown in the Immersive Reader.
+        // This basic example contains chunks of two different languages.
+        val chunk1 = hashMapOf("content" to irText1.text.toString(), "lang" to "en", "mimeType" to "text/plain")
+        val chunk2 = hashMapOf("content" to irText2.text.toString(), "lang" to "fr", "mimeType" to "text/plain")
+
+        val chunks = arrayListOf(HashMap<String, String>())
+        chunks.add(chunk1)
+        chunks.add(chunk2)
+
+        val content = hashMapOf("title" to irTitle.text, "chunks" to chunks)

```

Comment:
I think it's okay to keep Content and Chunk as classes. Options as a class too, but with parameter less constructors. When the sdk allows a parameter to be not given, please make sure the constructors don't require them (they can be overridden with setters)

Reply:
BTW, If you give default values to all the fields of a data class - an empty constructor is generated automatically by Kotlin. So I'll revert these to use data classes.